portal-api/api-tests/v1/gef/amendments/amendments.get.api-test.ts

```diff
@@ -0,0 +1,139 @@
+import app from '../../../../src/createApp';
+import testUserCache from '../../../api-test-users';
+
+import { MAKER } from '../../../../src/v1/roles/roles';
+import { getAmendmentUrl } from './get-amendment-url';
+import createApi from '../../../api';
+import { TestUser } from '../../../types/test-user';
+import { withRoleAuthorisationTests } from '../../../common-tests/role-authorisation-tests';
+import { withClientAuthenticationTests } from '../../../common-tests/client-authentication-tests';
+
+const { as, get } = createApi(app);
+
+const getPortalFacilityAmendmentMock = jest.fn() as jest.Mock<Promise<PortalFacilityAmendmentWithUkefId>>;
+
+jest.mock('../../../../src/v1/api', () => ({
+  ...jest.requireActual<AnyObject>('../../../../src/v1/api'),
+  getPortalFacilityAmendment: () => getPortalFacilityAmendmentMock(),
+}));
+
+const originalProcessEnv = { ...process.env };
+
+const validFacilityId = new ObjectId().toString();
+const validAmendmentId = new ObjectId().toString();
+
+const invalidId = 'invalid-id';
+
+describe('/v1/gef/facilities/:facilityId/amendments/:amendmentId', () => {
+  let testUsers: Awaited<ReturnType<typeof testUserCache.initialise>>;
+  let aMaker: TestUser;
+
+  describe('GET /v1/gef/facilities/:facilityId/amendments/:amendmentId', () => {
+    beforeEach(() => {
+      jest.resetAllMocks();
+    });
+
+    beforeAll(async () => {
+      testUsers = await testUserCache.initialise(app);
+      aMaker = testUsers().withRole(MAKER).one() as TestUser;
+    });
+
+    afterAll(() => {
+      jest.resetAllMocks();
+      process.env = originalProcessEnv;
+    });
+
+    describe('when FF_PORTAL_FACILITY_AMENDMENTS_ENABLED is disabled', () => {
+      beforeEach(() => {
+        process.env.FF_PORTAL_FACILITY_AMENDMENTS_ENABLED = 'false';
+      });
+
+      it(`should return a ${HttpStatusCode.NotFound} response`, async () => {
+        // Arrange
+        const url = getAmendmentUrl({ facilityId: validFacilityId, amendmentId: validAmendmentId });
+
+        // Act
+        const response = await as(aMaker).get(url);
+
+        // Assert
+        expect(response.status).toEqual(HttpStatusCode.NotFound);
+      });
+    });
+
+    describe('when FF_PORTAL_FACILITY_AMENDMENTS_ENABLED is enabled', () => {
+      beforeEach(() => {
+        process.env.FF_PORTAL_FACILITY_AMENDMENTS_ENABLED = 'true';
+      });
+
+      afterAll(() => {
+        jest.resetAllMocks();
+      });
+
+      withClientAuthenticationTests({
+        makeRequestWithoutAuthHeader: () => get(getAmendmentUrl({ facilityId: validFacilityId, amendmentId: validAmendmentId })),
+        makeRequestWithAuthHeader: (authHeader: string) =>
+          get(getAmendmentUrl({ facilityId: validFacilityId, amendmentId: validAmendmentId }), { headers: { Authorization: authHeader } }),
+      });
+
+      withRoleAuthorisationTests({
+        allowedRoles: [MAKER],
+        getUserWithRole: (role: Role) => testUsers().withRole(role).one() as TestUser,
```

Comment:
would love to make the `testUsers` typescript like I did in tfm-api - maybe worth looking at on the PUT or PATCH endpoint but leave for now :)

Reply:
Yep good shout, let's leave for now to get this in and defer to the next endpoint work